runelite-client/src/main/java/net/runelite/client/plugins/loginscreen/LoginScreenPlugin.java

```diff
@@ -245,59 +251,58 @@ private void overrideLoginScreen()
 	{
 		client.setShouldRenderLoginScreenFire(config.showLoginFire());
 
-		if (config.loginScreen() == LoginScreenOverride.OFF)
-		{
-			restoreLoginScreen();
-			return;
-		}
-
 		SpritePixels pixels = null;
-		if (config.loginScreen() == LoginScreenOverride.CUSTOM)
+		Random random = new Random(System.nanoTime());
+
+		switch (config.loginScreen())
 		{
-			if (CUSTOM_LOGIN_SCREEN_FILE.exists())
-			{
-				try
+			case OFF:
+				restoreLoginScreen();
+				return;
+			case CUSTOM:
+				if (CUSTOM_LOGIN_SCREEN_FILE.exists())
 				{
-					BufferedImage image;
-					synchronized (ImageIO.class)
-					{
-						image = ImageIO.read(CUSTOM_LOGIN_SCREEN_FILE);
-					}
+					pixels = getFileSpritePixels(CUSTOM_LOGIN_SCREEN_FILE);
+				}
+				break;
+			case RANDOM:
+				ArrayList<String> loginScreenOverrides = new ArrayList<>();
 
-					if (image.getHeight() > Constants.GAME_FIXED_HEIGHT)
+				for (LoginScreenOverride loginScreenOverride : LoginScreenOverride.values())
+				{
+					if (loginScreenOverride.getFileName() != null)
 					{
-						final double scalar = Constants.GAME_FIXED_HEIGHT / (double) image.getHeight();
-						image = ImageUtil.resizeImage(image, (int) (image.getWidth() * scalar), Constants.GAME_FIXED_HEIGHT);
+						loginScreenOverrides.add(loginScreenOverride.getFileName());
 					}
-					pixels = ImageUtil.getImageSpritePixels(image, client);
+				}
+
+				pixels = getClassPathFileSpritePixels(loginScreenOverrides.get(random.nextInt(loginScreenOverrides.size())));
+				break;
+
+			case RANDOM_CUSTOM:
+
+				try (Stream<Path> paths = Files.walk(CUSTOM_LOGIN_SCREEN_FOLDER.toPath()))
+				{
+					List<File> customScreens = paths.filter(Files::isRegularFile)
+						.filter(p -> p.getFileName().toString().toLowerCase().endsWith(".png") ||
+							p.getFileName().toString().toLowerCase().endsWith(".jpg") ||
+							p.getFileName().toString().toLowerCase().endsWith(".jpeg"))
+						.map(Path::toFile)
+						.collect(Collectors.toList());
+
+					File selectedCustomScreen = customScreens.get(random.nextInt(customScreens.size()));
+
+					pixels = getFileSpritePixels(selectedCustomScreen);
 				}
 				catch (IOException e)
 				{
 					log.error("error loading custom login screen", e);
```

Comment:
Is this accurate when `IOException` would be thrown by `Files.walk()`?

Reply:
I will do some testing on this.

Reply:
I'm not too sure what you mean by accurate but when IOException exception is thrown when "login-screen" folder is not found, therefore pixels will still be null so the following applies:

```
if (pixels != null)
{
	client.setLoginScreen(pixels);
}
```

My latest commit address the case that the "login-screen" folder exists but contains no image files.